Gems/Atom/RPI/Code/Tests/System/FeatureProcessorFactoryTests.cpp

```diff
@@ -98,32 +98,22 @@ namespace UnitTest
     {
         FeatureProcessorFactory::Get()->UnregisterFeatureProcessor<TestFeatureProcessorImplementation>();
 
-        // TestFeatureProcessorImplementation can no longer be created because it has been unregistered
-        FeatureProcessorPtr implementation1 = FeatureProcessorFactory::Get()->CreateFeatureProcessor(FeatureProcessorId{ TestFeatureProcessorImplementation::RTTI_TypeName() });
-        FeatureProcessorPtr implementation2 = FeatureProcessorFactory::Get()->CreateFeatureProcessor(FeatureProcessorId{ TestFeatureProcessorImplementation2::RTTI_TypeName() });
-
-        EXPECT_TRUE(implementation1 == nullptr);
-        EXPECT_TRUE(implementation2 != nullptr);
-    }
-
-    TEST_F(FeatureProcessorFactoryTests, UnregisterFeatureProcessor_MultipleImplmentationsOfTheSameInterface_OnlyTestFeatureProcessorImplementation2IsUnregistered)
-    {
-        FeatureProcessorFactory::Get()->UnregisterFeatureProcessor<TestFeatureProcessorImplementation2>();
-
-        // TestFeatureProcessorImplementation can no longer be created because it has been unregistered
+        // Should still be able to create both feature processors even with one unregistered.
         FeatureProcessorPtr implementation1 = FeatureProcessorFactory::Get()->CreateFeatureProcessor(FeatureProcessorId{ TestFeatureProcessorImplementation::RTTI_TypeName() });
         FeatureProcessorPtr implementation2 = FeatureProcessorFactory::Get()->CreateFeatureProcessor(FeatureProcessorId{ TestFeatureProcessorImplementation2::RTTI_TypeName() });
 
         EXPECT_TRUE(implementation1 != nullptr);
-        EXPECT_TRUE(implementation2 == nullptr);
+        EXPECT_TRUE(implementation2 != nullptr);
```

Comment:
EXPECT_NE should be used instead of EXPECT_TRUE when comparing multiple objects.
That way google test is able to resolve each parameter value when providing failure messages.

```suggestion
        EXPECT_NE(nullptr, implementation2);
```

When the EXPECT_TRUE condition fails, gtest can only output 
> "implementation2 != nullptr" has failed

as opposed to following when the relation EXPECT macros are used

> expected nullptr
> actual: 0x01...

Reply:
Agreed, I didn't touch this though because it already was this way. I'll go ahead and correct all the places in the FeatureProcessorFactoryTests since it's pretty small, but SceneTests is full of this problem as well and fixing would bloat this change considerably.